Source/Core/InputCommon/GCAdapter.cpp

```diff
@@ -351,7 +374,10 @@ static void Reset()
 
   if (s_adapter_thread_running.TestAndClear())
   {
-    s_adapter_thread.join();
+    s_rumble_data_available.Set();
+    s_adapter_input_thread.join();
+    s_adapter_output_thread.join();
+    s_rumble_data_available.Reset();
```

Comment:
The reset is not needed here (I thought this was fixed at some point?)

Other than that, LGTM after you squash all your commits together